scripts/logsConversion/logsToJsonConverter.py

```diff
@@ -0,0 +1,398 @@
+                        if self.log_content[-1].a:
+                            self.log_content.append(self.log_content[-1].copy())
+                        self.log_content[-1].a = param_dict
+
+                # regular parameters line on subsequent entry/entries (<key> = <value> <key> = <value>)
+                elif self.log_content and '=' in line:
+                    self.add_parameters(self.log_content[-1].a, line.strip())
+
+                else:
+                    pass
+                    # irrelevant line.
+
+        # write the no_space_in_message to a file for debugging purposes
+        with open('no_space.txt', 'w') as f:
+            for item in no_space_in_message:
+                f.write(f'{item} => {first_params[no_space_in_message.index(item)]}\n')
+
+    @staticmethod
+    def from_logs(path: str, node_name: str, output_path: str = OUTPUT_FOLDER):
+        result = LogsToJsonConverter(node_name, path, output_path)
+        log_content = load_json(path)
+        result.parse(log_content)
+
+
+def recursive_post_process_keys_for_statistics(txt: str, matches: list[tuple[str, str]], stats_dict: dict[str, str]) -> dict[str, str]:
+    if not matches:
+        return stats_dict
+
+    key, value = matches.pop()
+    key = key.strip()
+    value = value.strip() + (' ' + txt.strip() if txt else '')
+
+    txt = key[:3] if key.startswith('GB ') or key.startswith('MB ') or key.startswith('KB ') else key[:2] if key.startswith('B ') else ''
+    stats_dict[key.replace('GB ', '').replace('MB ', '').replace('KB ', '').replace('B ', '')] = value
+    recursive_post_process_keys_for_statistics(txt, matches, stats_dict)
+    return {k: stats_dict[k] for k in reversed(list(stats_dict.keys()))}
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='''
+        Runs node log conversion to JSON format. Example script:
+
+            python logsToJsonConverter --node_name=ovh-p03-validator-7 --path=logsPath/mx-chain-go-2024-12-10-10-14-21.log
+        ''',
+        epilog='\n',
+        formatter_class=argparse.RawTextHelpFormatter
+    )
+
+    parser.add_argument(
+        '--node_name',
+        required=True,
+        type=str,
+        help='Name of the node.'
+    )
+    parser.add_argument(
+        '--path',
+        required=True,
+        type=validate_file_path,
+        help='Path to the node logs file.'
+    )
+
+    args = parser.parse_args()
+
+    converter = LogsToJsonConverter(args.node_name)
+    with open(args.path, "r") as file:
+        log = file.readlines()
+        converter.parse(log)
+    output_file_name = args.path.split('/')[-1].replace('.log', '.jsonl')
+    output_file_path = converter.write_to_file(output_file_name)
+    print('Output file name: ', output_file_path)
+
+
+def validate_file_path(path: str):
+    if not os.path.isfile(path):
+        raise argparse.ArgumentTypeError(f"File '{path}' does not exist.")
```

Comment:
[`ArgumentTypeErrorArgumentTypeError
`](https://docs.python.org/3/library/argparse.html#argparse.ArgumentTypeError) is not a good match here, I think.